deluge/core/torrentmanager.py

```diff
@@ -1339,6 +1340,48 @@ def on_alert_torrent_checked(self, alert):
 
         torrent.update_state()
 
+    def on_alert_torrent_need_cert(self, alert):
+        """Alert handler for libtorrent torrent_need_cert_alert"""
+
+        if not self.config['ssl_peers']['enabled']:
+            return
+
+        torrent_id = str(alert.handle.info_hash())
+        base_path = self.config['ssl_peers']['certificate_location']
+        if not os.path.isdir(base_path):
+            return
+
+        certificate_path = None
+        private_key_path = None
+        dh_params_path = None
+        for file_name in [torrent_id + '.dh', 'default.dh']:
+            params_path = os.path.join(base_path, file_name)
+            if os.path.isfile(params_path):
+                dh_params_path = params_path
+                break
+        if dh_params_path:
+            for file_name in [torrent_id, 'default']:
+                crt_path = os.path.join(base_path, file_name)
+                key_path = crt_path + '.key'
+                if os.path.isfile(crt_path) and os.path.isfile(key_path):
+                    certificate_path = crt_path
+                    private_key_path = private_key_path
+                    break
```

Comment:
Is there a consensus on file naming convention here? I notice that the cert file is missing an extension which doesn't seem quite right.

I feel that the files should end in `.pem` to indicate the encoding. The [libtorrent tests](https://github.com/arvidn/libtorrent/blob/5ce4c516479085eda9327d35891b7732d6d5037b/examples/client_test.cpp#L1067) use:
* dhparams.pem
* <torrent_id>.pem
* <torrent_id>_key.pem

Also these default files will need documented. 

I wonder if we can make the code cleaner with pathlib and for else:

 ```suggestion
        certs_dir = Path(self.config['ssl_torrents_certs'])
        dh_params_file = certs_dir / f'{torrent_id}.dh'
        if not dh_params.is_file():
             dh_params_file = certs_dir / 'default.dh'
   
        for filename in [torrent_id, 'default']:
            cert_file = certs_dir / f'{filename}.pem'
            priv_key_file = certs_dir / f'{filename}_key.pem'
            if dh_params.is_file() and cert.is_file() and priv_key.is_file():
                break
        else:
            log.error('<Certs not found...>')
            return
                   
```
In fact this feels like reusable code for a common helper function